core/src/main/java/org/mule/runtime/core/api/util/ClassUtils.java

```diff
@@ -372,36 +379,96 @@ public static Class loadClass(final String className, final ClassLoader classLoa
-    Class<?> clazz = target.getClass();
-    Field field;
-    while (!Object.class.equals(clazz)) {
-      try {
-        field = clazz.getDeclaredField(fieldName);
-        field.setAccessible(true);
-        field.set(target, value);
+    Field f = getField(target.getClass(), fieldName, recursive);
+    f.setAccessible(true);
+    return (T) f.get(target);
+  }
 
-        return;
-      } catch (NoSuchFieldException e) {
-        // ignore and look in superclass
-        if (recursive) {
-          clazz = clazz.getSuperclass();
-        } else {
-          break;
-        }
-      }
+  /**
+   * Gets a static field value of a given class
+   * @param targetClass the class that holds the requested static field.
+   * @param fieldName the name of the field
+   * @param recursive flag to lookup for the field in subclasses or not
+   * @param <T> the type of the field
+   * @return the value of the field
+   * @throws NoSuchFieldException when the field is not found
+   * @throws IllegalAccessException when the field is not reachable.
+   */
+  public static <T> T getStaticFieldValue(Class<?> targetClass, String fieldName, boolean recursive)
+      throws NoSuchFieldException, IllegalAccessException {
+    Field field = getField(targetClass, fieldName, recursive);
+    if (!Modifier.isStatic(field.getModifiers())) {
+      throw new IllegalAccessException(String.format("The %s field of %s class is not static", fieldName, targetClass.getName()));
     }
+    field.setAccessible(true);
+    return (T) field.get(null);
+  }
 
-    throw new NoSuchFieldException(String.format("Could not find field '%s' in class %s", fieldName,
-                                                 target.getClass().getName()));
+  /**
+   * Sets a field of an object with the given value. If this is a final field, there will be
+   * an attempt to update the value.
+   * Notice: If the field is final and it was initialized using a constant the value change
+   * may not be reflected in due compiler optimizations.
+   * http://java.sun.com/docs/books/jls/third_edition/html/memory.html#17.5.3
+   * @param target the object that holds the target field
+   * @param fieldName the name of the field
+   * @param value the value to set
+   * @param recursive flags to lookup the field in subclasses of the target object
+   * @throws IllegalAccessException the field is not reachable
+   * @throws NoSuchFieldException the field does not exists.
+   */
+  public static void setFieldValue(Object target, String fieldName, Object value, boolean recursive)
+      throws IllegalAccessException, NoSuchFieldException {
+    Field field = getField(target.getClass(), fieldName, recursive);
+    field.setAccessible(true);
+    if (Modifier.isFinal(field.getModifiers())) {
+      Field modifiersField = getField(Field.class, "modifiers", false);
+      modifiersField.setAccessible(true);
+      modifiersField.setInt(field, field.getModifiers() & ~Modifier.FINAL);
```

Comment:
set accessible and modifiers to their original value after getting the value of the field, in a finally block